portal-impl/src/com/liferay/portal/security/permission/StagingPermissionChecker.java

```diff
@@ -46,16 +46,8 @@ public static Long getGroupId() {
 		return groupId;
 	}
 
-	public static void removeGroupId() {
-		if (_log.isDebugEnabled()) {
-			_log.debug("removeGroupId ");
-		}
-
-		_groupId.remove();
-	}
-
 	public static void setGroupId(Long groupId) {
-		_groupId.set(groupId);
+		_groupId.setWithSafeClosable(groupId);
```

Comment:
Hi @ChrisKian,

First of all, sorry for the late reply. 

Maybe I am missing something but, how are returning the original value back to groupId if we are not colling the method close of SafeClosable?

Thanks.

Reply:
The setWithSafeCloseable method doesn't actually require any closing for the initial value to be reset.  It should be be set for the duration of the context, so for example placing the setWithSafeCloseable in a try block will set the value throughout the try block, but implicitly reset during the finally block.

So, if we call the setWithSafeCloseable method here with groupId X, the _groupId variable will remain X until we return from the method (after line 164).

Reply:
Hi @chrisKian

As far as I know, the method close is only called automatically when it is used in a try with resources statement:
https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html

> So, if we call the setWithSafeCloseable method here with groupId X, the _groupId variable will remain X until we return from the method (after line 164).

So, are you saying that the method close is called after the return statement in _hasPermission_? Do you know how this happen? Sorry but I always had to return the SafeCloseable and use it in a try with resources to close it automatically,

Thanks.

Reply:
It may need to be embedded in a try block; I'll push those changes now